MovieViewer/MoviesSearchViewController.swift

```diff
@@ -258,42 +260,35 @@ extension MoviesSearchViewController: UITableViewDataSource {
         return [shareAction, favoriteAction]
     }
     
-    func checkFavorite(row: Int) -> Bool {
-        
-        var flag = false
-        // get movie id
-        if let id = self.filteredMovies![row].id {
-            if self.favoriedMovies.contains(id) {
-                flag = true
-            }
-        }
-        return flag
-    }
-    
-    func addFavoriteMovie(id: Int) {
+    func retreiveFavoritedMovies() {
         
-        do {
-            try realm.write() {
-                let favoritedMovie = FavoritedMovie()
-                favoritedMovie.id = id
-                realm.add(favoritedMovie)
+        ref.child("anonymous/favorited").observeEventType(.Value, withBlock: { (snapshot: FIRDataSnapshot) in
```

Comment:
yup! this is correct! thanks for figuring this out. it's a little bit "heavy" in a way, but this is basically what you'd like to do. 

i think a slightly better way would be to observe child events, such as `ChildChanged`. we can talk about that more today too.